modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AttachmentAwareClient.java

```diff
@@ -77,12 +82,54 @@ private T uploadAndAttachAttachment(T itemToModify, Path fileToAttach, HttpEntit
             Validate.validState(response.getBody() != null);
             return response.getBody();
         } catch (HttpClientErrorException | HttpServerErrorException e) {
-            LOGGER.warn("Request to get attach {} to {} failed with {}", fileToAttach, self, e.getStatusCode());
+            LOGGER.warn("Request to attach {} to {} failed with {}", fileToAttach, self, e.getStatusCode());
             LOGGER.debug("Error: ", e);
             return itemToModify;
         } catch (ExecutionException e) {
             LOGGER.warn("Request to attach {} to {} failed with {}", fileToAttach, self, e.getMessage());
             return itemToModify;
         }
     }
+
+    public List<SW360SparseAttachment> getReleaseAttachments(T itemToModify, HttpHeaders header) {
+        final String self = itemToModify.get_Links().getSelf().getHref();
+        try {
+            ResponseEntity<Resource<SW360AttachmentList>> response = doRestGET(self + ATTACHMENTS_ENDPOINT, header,
+                    new ParameterizedTypeReference<Resource<SW360AttachmentList>>() {});
+            checkRestStatus(response);
+            Validate.validState(response.getBody() != null);
+            return getSw360SparseAttachments(response);
+        } catch (HttpServerErrorException e) {
+            LOGGER.debug("Request to get attachments from {} failed with {}", self, e.getStatusCode());
+            LOGGER.debug("Error: ", e);
+            return new ArrayList<>();
+        } catch (ExecutionException e) {
+            LOGGER.warn("Request to get attachments from {} failed with {}", self, e.getMessage());
+            return new ArrayList<>();
+        }
+    }
+
+    public Optional<Path> downloadAttachment(String itemHref, SW360SparseAttachment attachment, Path downloadPath, HttpHeaders header) {
+        String attachmentId = attachment.getAttachmentId();
+        String url = itemHref + "/attachments/" + attachmentId;
+        try {
+            HttpEntity<String> httpEntity = new HttpEntity<>(header);
+            ResponseEntity<byte[]> response = restTemplate.exchange(url, HttpMethod.GET, httpEntity, byte[].class);
+            checkRestStatus(response);
+            Validate.validState(response.getBody() != null);
+
+            Optional<byte[]> body = Optional.of(response.getBody());
+
+            return Optional.of(Files.write(downloadPath.resolve(attachment.getFilename()), body.get()));
+        } catch (HttpClientErrorException e) {
+
+            LOGGER.debug("Request to get attachment {} from {} failed with {}", attachmentId, itemHref, e.getStatusCode());
```

Comment:
I would like to add, that the general handling here and in the method above is a bit weird. Not being able to provide the attachment is in general a problem because we will not be able to generate things, e.g., the source bundle properly. All of these exceptions should result in a warning at least. We should also always log the exception stack trace as debug information, that will help us if we encounter the issue in a support call (see below).